kornia/feature/hardnet.py

```diff
@@ -165,5 +166,5 @@ def forward(self, input: torch.Tensor) -> torch.Tensor:
         x_norm: torch.Tensor = self._normalize_input(input)
         x_features: torch.Tensor = self.features(x_norm)
         x_prePCA = F.normalize(x_features.view(x_features.size(0), -1))
-        pca = torch.mm(x_prePCA - self.mean, self.components.T)
+        pca = torch.mm(x_prePCA - self.mean, self.components)
```

Comment:
@ducha-aiki can we have an inhouse pca near future ? :)

Reply:
So far learning is done by sklearn, the only inference is torch. But - yes, why not?